File: lab2/zadanie2/zadanie2.py
from math import sqrt

def czy_pierwsza(x):
	if x <= 1:
		return False
	elif x == 2:
		return True
	elif x % 2 == 0:
		return False
	for i in range(3, int(sqrt(x)+1)):
		if x % i == 0:
			return False
	return True

def main():
	liczba = int(input("Podaj liczbę: "))
	if czy_pierwsza(liczba):
		print("Liczba jest pierwsza")
	else:
		print("Liczba nie jest pierwsza")

if __name__ == "__main__":
	main()

File: lab2/zadanie2/zadanie2.c
#include <math.h>
#include <stdio.h>

int liczba;

int czy_pierwsza(int liczba){
	if (liczba <= 1){
		return 0;
	}
	else if (liczba == 2){
		return 1;
	}
	else if (liczba%2 == 0){
		return 0;
	}
	for (int i=3; i <= sqrt(liczba); i++){
		if (liczba%i==0){
			return 0;
		}
	}
	return 1;
}

int main(){
	printf("Podaj liczbę: ");
	scanf("%d", &liczba);
	if (czy_pierwsza(liczba)){
		printf("Liczba jest pierwsza");
	}
	else {
		printf("Liczba nie jest pierwsza");
	}


}
